### PR TITLE
AJP connector defaults fixes and typos fixes

### PR DESCRIPTION
**Pull request by thanosploumis**
I made one typo mistake during previous push.It needs to be fixed.
Also during refactoring noe-tests based on noe-core refactoring AJP connector is failing (with portOutOfIndex error) during tomcat start as it has no specified port. Even it is specified, if secretRequired is absent the default is true.So i set it during httpsSecure setting to false.If the user/tester wants to change it, he can change it afterwards with specific AJP connector update

@jstefl FYI

### CODE REVIEW

**Comment by jstefl** [comment]
> Also during refactoring noe-tests based on noe-core refactoring AJP connector is failing (with portOutOfIndex error) during tomcat start as it has no specified port

Contract of `AjpConnectorTomcat` says "no defaults are provided", so I would be very cautious. 
What happens exactly, how the failing code looks like? 


> Even it is specified, if secretRequired is absent the default is true.

That is standard Tomcat behavior, so it seems to me that it is [correct](https://tomcat.apache.org/tomcat-7.0-doc/config/ajp.html). I think I need more of the context - the failing code/data will do.


> So i set it during httpsSecure setting to false. If the user/tester wants to change it, he can change it afterwards with specific AJP connector update

I am not sure that I am following. A problematic code sample should help.

**Comment by thanosploumis** [comment]
> Contract of `AjpConnectorTomcat` says "no defaults are provided", so I would be very cautious.
> What happens exactly, how the failing code looks like?
 For example if inside server.xml,
`<Connector protocol="AJP/1.3" redirectPort="8080"/>`
tomcat fail to start with 
'Failed to initialize component [Connector[AJP/1.3-auto-1]]
Caused by: java.lang.IllegalArgumentException: port out of range:-1`
This because there is no "port" attribute. If i place some random "port" attribute, tomcat passes the error about port and wait for some secret.
> That is standard Tomcat behavior, so it seems to me that it is [correct](https://tomcat.apache.org/tomcat-7.0-doc/config/ajp.html). I think I need more of the context - the failing code/data will do.
If on previous connector, set secretRequired="true", tomcat waits for a secret attribute.But during automatic AJP creation during HttpsConnector we dont set any secret. Thats why I mentioned, that attribute secretRequired should be false in order to tomcat start.And it starts indeed if so.
To sum up during AJP connector set after SecureHttpsConnector creating, shouldn't we make ajp connector as default as possible inside server.xml in order not to fail tomcat wihout reason?
Otherwise user/client should add code like this:
`        tconf.ajpConnector(new AjpConnectorTomcat().setPort(num).setSecretRequired(false))`
> I am not sure that I am following. A problematic code sample should help.

**Comment by thanosploumis** [comment]
> Contract of `AjpConnectorTomcat` says "no defaults are provided", so I would be very cautious.
> What happens exactly, how the failing code looks like?
>> For example if inside server.xml,
`<Connector protocol="AJP/1.3" redirectPort="8080"/>`
tomcat fail to start with 
'Failed to initialize component [Connector[AJP/1.3-auto-1]]
Caused by: java.lang.IllegalArgumentException: port out of range:-1`
This because there is no "port" attribute. If i place some random "port" attribute, tomcat passes the error about port and wait for some secret.
> That is standard Tomcat behavior, so it seems to me that it is [correct](https://tomcat.apache.org/tomcat-7.0-doc/config/ajp.html). I think I need more of the context - the failing code/data will do.
If on previous connector, set secretRequired="true", tomcat waits for a secret attribute.But during automatic AJP creation during HttpsConnector we dont set any secret. Thats why I mentioned, that attribute secretRequired should be false in order to tomcat start.And it starts indeed if so.
To sum up during AJP connector set after SecureHttpsConnector creating, shouldn't we make ajp connector as default as possible inside server.xml in order not to fail tomcat wihout reason?
Otherwise user/client should add code like this:
`        tconf.ajpConnector(new AjpConnectorTomcat().setPort(num).setSecretRequired(false))`
> I am not sure that I am following. A problematic code sample should help.

**Comment by thanosploumis** [comment]
> >Contract of `AjpConnectorTomcat` says "no defaults are provided", so I would be very cautious.
> >What happens exactly, how the failing code looks like?
>For example if inside server.xml,
>`<Connector protocol="AJP/1.3" redirectPort="8080"/>`
>tomcat fail to start with 
>'Failed to initialize component [Connector[AJP/1.3-auto-1]]
>Caused by: java.lang.IllegalArgumentException: port out of range:-1`
>This because there is no "port" attribute. If i place some random "port" attribute, tomcat passes the error about port and wait for >some secret.
> >That is standard Tomcat behavior, so it seems to me that it is [correct](https://tomcat.apache.org/tomcat-7.0->>doc/config/ajp.html). I think I need more of the context - the failing code/data will do.
>If on previous connector, set secretRequired="true", tomcat waits for a secret attribute.But during automatic AJP creation during >HttpsConnector we dont set any secret. Thats why I mentioned, that attribute secretRequired should be false in order to tomcat >start.And it starts indeed if so.
>To sum up during AJP connector set after SecureHttpsConnector creating, shouldn't we make ajp connector as default as >possible inside server.xml in order not to fail tomcat wihout reason?
>Otherwise user/client should add code like this:
>`        tconf.ajpConnector(new AjpConnectorTomcat().setPort(num).setSecretRequired(false))`

**Comment by thanosploumis** [comment]
> Contract of `AjpConnectorTomcat` says "no defaults are provided", so I would be very cautious.
> What happens exactly, how the failing code looks like?

For example, if inside server.xml,
`<Connector protocol="AJP/1.3" redirectPort="8080"/>`
tomcat fails to start with 
`Failed to initialize component [Connector[AJP/1.3-auto-1]]
Caused by: java.lang.IllegalArgumentException: port out of range:-1`
This because there is no "port" attribute. If I place some random "port" attribute, tomcat is passing the error about port and wait for some secret.

> That is standard Tomcat behavior, so it seems to me that it is [correct](https://tomcat.apache.org/tomcat-7.0-> doc/config/ajp.html). I think I need more of the context - the failing code/data will do.

If on the previous connector, set secretRequired="true", tomcat's waiting for a secret attribute. But during automatic AJP creation during HttpsConnector we don't set any secret. Thats why I mentioned, that attribute secretRequired should be false in order to tomcat start. And it starts indeed if so.
To sum up during the AJP connector set after SecureHttpsConnector creating, shouldn't we make AJP connector as default as possible inside server.xml in order not to fail tomcat without reason?
Otherwise user/client should add code like this:
`        tconf.ajpConnector(new AjpConnectorTomcat().setPort(num).setSecretRequired(false))`

**Comment by jstefl** [comment]
Could you share the spcific code what creates the :
````
<Connector protocol="AJP/1.3" redirectPort="8080"/>
`````

**Comment by thanosploumis** [comment]
> Could you please share the specific code what creates the :
> 
> ```
> <Connector protocol="AJP/1.3" redirectPort="8080"/>
>```

`configurator.httpsConnector(new SecureHttpConnectorTomcat().setPort(8080))`

Basically the httpsConnector creates one HTTP and one AJP for redirection reasons

**Comment by jstefl** [comment]
I see. Now I understand. KISS principle was broken and here we are...

The key problem is that the configurator tries to be too clever (automagically) and this is the result. 
It tries to set redirect port automatically but in real world we don't know whether this is correct (for default config - maybe).

The correct way here would be to get rid of automagical redirect port configuration and let the client/tester do the whole proper configuration as they need - because only they should know what is expected - they have full context.

I would consider fixes reflecting the above, true is that it means investigating what that means for unit /product test and do the changes.

WDYT?

**Comment by thanosploumis** [comment]
If we remove redirectionPort it will remain `<Connector protocol="AJP/1.3">. Pretty unnessecary/empty. 
Basically the main problem is creating an incomplete AJP connector, not the redirectPort attribute alone

Maybe you meant the whole AJP connector right?

**Comment by jstefl** [comment]
> Maybe you meant the whole AJP connector right?

Sure, yes I meant.

**Comment by thanosploumis** [comment]
Affirmative :) I will adjust tests as well

**Comment by jstefl** [comment]
good, good

File: core/src/main/groovy/noe/tomcat/configure/ConnectorAttributesTransformer.groovy [hunk 1]
```diff
@@ -169,7 +169,7 @@ class ConnectorAttributesTransformer {
       if (connector.getSslPassword() != null && !connector.getSslPassword().isEmpty()) {
         attributes.put('SSLPassword', connector.getSslPassword())
       }
-      if (connector.getSslEnabledProtocols() != null && connector.getSslEnabledProtocols()) {
+      if (connector.getSslEnabledProtocols() != null) {
```
Comment (jstefl) [inline core/src/main/groovy/noe/tomcat/configure/ConnectorAttributesTransformer.groovy:172]: Why?

File: core/src/main/groovy/noe/tomcat/configure/AjpConnectorTomcat.groovy [hunk 2]
```diff
@@ -19,10 +19,6 @@ public class AjpConnectorTomcat extends ConnectorTomcatAbstract<AjpConnectorTomc
   private String secret
   private String allowedRequestAttributesPattern
 
-  public AjpConnectorTomcat() {
```
Comment (jstefl) [inline core/src/main/groovy/noe/tomcat/configure/AjpConnectorTomcat.groovy:22]: I would leave protocol = "AJP/1.3" because this is in AjpConnectorTomcat class and there is no other option so far (AFAIK), si this is IMHO OK.

File: core/src/main/groovy/noe/tomcat/configure/ConnectorAttributesTransformer.groovy (hunk 1 above)
Comment (thanosploumis) [reply core/src/main/groovy/noe/tomcat/configure/ConnectorAttributesTransformer.groovy:172]: If user tried to set this value to `false` it will never be changed. I'm sure I placed this part there by wrong copy-paste

File: core/src/main/groovy/noe/tomcat/configure/AjpConnectorTomcat.groovy (hunk 2 above)
Comment (thanosploumis) [reply core/src/main/groovy/noe/tomcat/configure/AjpConnectorTomcat.groovy:22]: It sure doesn't affect anything. Will put it back

**Comment by thanosploumis** [comment]
Tested here https://jws-jbcs-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/noe/job/noe-core/job/noe-core-integration-linux-clone-aploumis/37/

File: core/src/main/groovy/noe/tomcat/configure/ConnectorAttributesTransformer.groovy (hunk 1 above)
Comment (jstefl) [reply core/src/main/groovy/noe/tomcat/configure/ConnectorAttributesTransformer.groovy:172]: I see, good point.

**Review by jstefl — APPROVED** [review]
